src/components/theme/Theme.tsx

```diff
@@ -43,62 +48,48 @@ const ThemePage = () => {
   const navigate = useNavigate();
   const { MAIN } = ROUTE_PATHS;
   const [productList, setProductList] = useState<ThemeProducts["list"]>([]);
-  const [cursor, setCursor] = useState(0);
-  const [hasMore, setHasMore] = useState(true);
-  const [isLoading, setIsLoading] = useState(false);
-
   const observerRef = useRef<HTMLDivElement | null>(null);
 
+  const { data, fetchNextPage, hasNextPage, isFetchingNextPage } =
+    useInfiniteQuery({
+      queryKey: ["getThemeProducts"],
+      queryFn: ({ pageParam = 0 }) =>
+        getThemeProducts(themeId as string, pageParam),
+      initialPageParam: 0,
+      getNextPageParam: (lastPage) => {
+        const { cursor, hasMoreList } = lastPage.data.data;
+        return hasMoreList ? cursor + 10 : undefined;
+      },
+    });
+
   useEffect(() => {
+    if (!data) return;
+    setProductList(data.pages.flatMap((page) => page.data.data.list));
+  }, [data]);
```

Comment:
데이터 가공을 위해 별도로 추가하신 state는 useMemo를 활용하면 더 깔끔하게 관리할 수 있을 것 같아요!
특히 useEffect를 사용해 state를 따로 관리하기보다는, 계산 가능한 값이라면 useMemo로 처리하는 것이 코드도 간결해지고, 사이드 이펙트도 줄일 수 있어요.
```suggestion
  const productList = useMemo(
    () => data?.pages.flatMap((page) => page.data.data.list) ?? [],
    [data],
  );
```